### PR TITLE
add objective-c GDPR sample

### PR DESCRIPTION
**Pull request by kbataran**
iOS objective-C sample for GDPR

### CODE REVIEW

File: gdpr/Podfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
```
Comment (abashyam) [inline gdpr/Podfile]: I realize this is commented but we should identify 8.0 which is the minimum we support.

File: gdpr/Podfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+target 'gdpr' do
+  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
+  # use_frameworks!
```
Comment (abashyam) [inline gdpr/Podfile]: I thought we always needed this when we use frameworks. Is my understanding wrong. We should only be using dynamic frameworks.

**Comment by Mriegel** [comment]
I think we need to build this out a little more.  I think there should be a screen past the login page with a few buttons for logout and opt out to demonstrate how to do that properly (similar to [Android](https://github.com/localytics/android-samples/blob/master/SDK-v5/gdpr/app/src/main/java/com/localytics/gdpr/MainActivity.java#L55-L65))

**Comment by kbataran** [comment]
@Mriegel Good point- I've added a home screen and buttons to opt out/in and logout

File: gdpr/gdpr/AppDelegate.m [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    
+    [Localytics pauseDataUploading:YES];
+    [Localytics autoIntegrate:@"a39f6f0cb378dbd8239767d-487f3e04-8691-11e6-ab4e-0098ae15fcaa"
+        withLocalyticsOptions:@{
+                                LOCALYTICS_WIFI_UPLOAD_INTERVAL_SECONDS: @5,
+                                LOCALYTICS_GREAT_NETWORK_UPLOAD_INTERVAL_SECONDS: @10,
+                                LOCALYTICS_DECENT_NETWORK_UPLOAD_INTERVAL_SECONDS: @30,
+                                LOCALYTICS_BAD_NETWORK_UPLOAD_INTERVAL_SECONDS: @90
+                                }
+                launchOptions:launchOptions];
```
Comment (Mriegel) [inline gdpr/gdpr/AppDelegate.m:30]: You need to check if the user is logged in here.  If they are logged in then you should be checking their opt in status on a server.

**Review by abashyam — COMMENTED** [review]
we shouldn't be adding  gdpr/Pods/Target Support Files to the repo. a Pod install should be able to set this up.

File: gdpr/Podfile (hunk 2 above)
Comment (abashyam) [reply gdpr/Podfile]: xcuserdata is personal to a developer and should not be under source control.

File: gdpr/gdpr/AppDelegate.m [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    
+    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"isLoggedIn"]) {
+        //TODO check user's opt-out status on your server
+        BOOL optOutStatus = NO;
+        [Localytics setPrivacyOptedOut:optOutStatus];
+        //go to home screen
+        self.window.rootViewController = [self.window.rootViewController.storyboard instantiateViewControllerWithIdentifier:@"HomeVC"];
```
Comment (abashyam) [inline gdpr/gdpr/AppDelegate.m:37]: Customers would want their app to be useable even when the user has not opted out. So the rootViewController should be rendered and the flow of login shown. cannot do authentication without an UI.

File: gdpr/gdprUITests/gdprUITests.m [hunk 5]
```diff
@@ -0,0 +1,40 @@
+//
```
Comment (abashyam) [inline gdpr/gdprUITests/gdprUITests.m:1]: I prefer to add the entire opt out flow in a Test so it makes our testing in future releases easier. If we feel like it's not needed lets remove the test target and associated files

File: gdpr/gdpr/AppDelegate.m (hunk 4 above)
Comment (kbataran) [reply gdpr/gdpr/AppDelegate.m:37]: not sure i understand your comment, why would I show the login viewcontroller if the user is already logged in?

**Review by Mriegel — APPROVED** [review]
We should squash and merge here

File: gdpr/gdpr/AppDelegate.m (hunk 4 above)
Comment (abashyam) [reply gdpr/gdpr/AppDelegate.m:37]: self.window.rootViewController = [self.window.rootViewController.storyboard instantiateViewControllerWithIdentifier:@"HomeVC"];
 Is within an if what happens in the else case?

Comment (kbataran) [reply gdpr/gdpr/AppDelegate.m:37]: Not sure what you mean by this ^ Can you explain?

**Review by abashyam — APPROVED** [review]
Customers use our samples to understand how to better upgrade their apps. To Better assist this effort, we should enhance the app in the near future to cover the case of what should be done when the app allows functionality when the user is not logged in. Including offering prompts for privacy etc.

**Pull request merged**